client/src/pages/SubmitDesign/SubmitDesign.tsx

```diff
@@ -1,33 +1,78 @@
+import { useState, ChangeEvent } from 'react';
+import { useLocation } from 'react-router-dom';
 import CssBaseline from '@material-ui/core/CssBaseline';
 import Paper from '@material-ui/core/Paper';
 import Button from '@material-ui/core/Button';
 import Grid from '@material-ui/core/Grid';
+import Box from '@material-ui/core/Box';
 import Typography from '@material-ui/core/Typography';
 import AuthHeader from '../../components/AuthHeader/AuthHeader';
 import useStyles from './useStyles';
 import CloudUploadIcon from '@material-ui/icons/CloudUpload';
+import { uploadSubmissionPic, createSubmission } from '../../helpers/APICalls/submission';
 
 export default function SubmitDesign(): JSX.Element {
+  const [allPics, setAllPics] = useState<string[]>([]);
   const classes = useStyles();
+  const location = useLocation();
+  const contestId = location.pathname.slice(15);
```

Comment:
You want to avoid this, if the pathname changes in length you will get an incorrect contest id.
Look into the usage of useParams()

Ref: https://medium.com/geekculture/how-to-use-react-router-useparams-436851fd5ef6

Reply:
I tried to use useParams but typescript was saying that it was always possibly undefined.  I can look into that again to see if I can get it to play nice with typescript. I'll do that today.

Reply:
This is fixed. (with useParams)